[Security Solution] Create alert only index

christineweng opened the pull request:
## Summary

This PR creates an alert only index along with default security data view when app starts. When the `newDataViewPickerEnabled` flag is enabled, alerts page uses alert only data view, vs when the flag is off, alerts page relies on sourcerer and the default data view is used. 

![image](https://github.com/user-attachments/assets/ad617a09-e8f4-4d38-b6d5-99a31e791a99)

![image](https://github.com/user-attachments/assets/513d1827-0ccd-43d8-ac8f-d3ce4baa0a5e)


The number of fields in field browser changes because we only fetch alert index fields now

|Flag enabled - using data view picker | Flag disabled - using sourcerer |
| ------ | ----- |
|![image](https://github.com/user-attachments/assets/6ef0d80a-fb70-43fe-b410-f90d3ee0b393)|![image](https://github.com/user-attachments/assets/5d85dba7-2f36-455f-a423-3f9ec78371e8)|



### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)




<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

christineweng committed:
alert only data view creation

christineweng commented on x-pack/solutions/security/plugins/security_solution/public/sourcerer/containers/use_signal_helpers.tsx:
```diff
@@ -85,6 +85,7 @@ export const useSignalHelpers = (): {
           // first time signals is defined and validated in the sourcerer
           // redo indexFieldsSearch
           indexFieldsSearch({ dataViewId: sourcererDataView.defaultDataView.id });
+          // indexFieldsSearch({ dataViewId: sourcererDataView.alertDataView.id });
```
added to account for index search but commenting out has no effect...

PhilippeOberti replied:
What do you mean by `commenting out has no effect`? Do we still need this? Or do we need to understand why it has no effect?

elasticmachine commented:
Pinging @elastic/security-threat-hunting-investigations (Team:Threat Hunting:Investigations)

lgestc commented on x-pack/solutions/security/plugins/security_solution/public/data_view_manager/constants.ts:
```diff
@@ -6,6 +6,7 @@
  */
 
 export const DEFAULT_SECURITY_SOLUTION_DATA_VIEW_ID = 'security-solution-default';
+export const DEFAULT_SECURITY_SOLUTION_ALERT_DATA_VIEW_ID = 'security-solution-alert';
```
can we reuse the common const defined in `x-pack/solutions/security/plugins/security_solution/common/constants.ts`? :)

lgestc reviewed (approved):
looking good to me, does it mean we could potentially get rid of the `signal` related portion of the state and just use this data view everywhere?

PhilippeOberti commented on x-pack/solutions/security/plugins/security_solution/public/data_view_manager/redux/slices.ts:
```diff
@@ -44,6 +45,9 @@ export const sharedDataViewManagerSlice = createSlice({
     setDefaultDataViewId: (state, action: PayloadAction<string>) => {
       state.defaultDataViewId = action.payload;
     },
+    setAlertDataViewId: (state, action: PayloadAction<string>) => {
```
See my previous suggestion to update the `setDefaultDataViewId` to `setDataViewIds` (or something like that)

PhilippeOberti reviewed (commented):
Left a few comments, none as blockers, more like suggestions and questions :)

PhilippeOberti commented:
Also @christineweng, do we want to update the AI4DSOC alert summary page code [here](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/public/detections/components/alert_summary/wrapper.tsx#L67) to point to the new security alert dataView?

christineweng committed:
PR comments

christineweng committed:
rename label to sentence case

christineweng commented:
> looking good to me, does it mean we could potentially get rid of the `signal` related portion of the state and just use this data view everywhere?

@lgestc yes to the `signalIndexName`, we can just call `getIndexPattern` on the detections data view, not sure about the boolean though

christineweng commented:
> Also @christineweng, do we want to update the AI4DSOC alert summary page code [here](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/public/detections/components/alert_summary/wrapper.tsx#L67) to point to the new security alert dataView?

@PhilippeOberti thanks for the review and feedback! all very good points. Because the `detections` scope now points to the alert data view by default, no changes needed on the AI4FSOC wrappers, I've removed the to do comments and updated that `signalIndexName` to pull from the new data view.

PhilippeOberti reviewed (approved):
Thanks for making all these changes!! LGTM for the @elastic/security-threat-hunting-investigations team :)

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311723)
* Commit: 0f5482ffa9c0600b8cb6454437389f82d700e4ad

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/59f635c9c86ce178e5fb79ba2121d86398af922c) | [after](https://github.com/elastic/kibana/commit/0f5482ffa9c0600b8cb6454437389f82d700e4ad) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.4MB | 9.4MB | +1.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/59f635c9c86ce178e5fb79ba2121d86398af922c) | [after](https://github.com/elastic/kibana/commit/0f5482ffa9c0600b8cb6454437389f82d700e4ad) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 95.4KB | 95.4KB | +40.0B |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/59f635c9c86ce178e5fb79ba2121d86398af922c) | [after](https://github.com/elastic/kibana/commit/0f5482ffa9c0600b8cb6454437389f82d700e4ad) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 625 | 626 | +1 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/59f635c9c86ce178e5fb79ba2121d86398af922c) | [after](https://github.com/elastic/kibana/commit/0f5482ffa9c0600b8cb6454437389f82d700e4ad) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 343 | 345 | +2 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/59f635c9c86ce178e5fb79ba2121d86398af922c) | [after](https://github.com/elastic/kibana/commit/0f5482ffa9c0600b8cb6454437389f82d700e4ad) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 723 | 724 | +1 |


</details>


### History
* :green_heart: [Build #311082](https://buildkite.com/elastic/kibana-pull-request/builds/311082) succeeded 0d4303935c373da0f0d529064c826ccfe066c4ff

cc @christineweng

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311723","number":311723,"commit":"0f5482ffa9c0600b8cb6454437389f82d700e4ad"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311082","number":311082,"commit":"0d4303935c373da0f0d529064c826ccfe066c4ff"}],"number":311723}
buildkite-pr-comment-->

andrew-goldstein reviewed (approved):
## Files by Code Owner

### elastic/security-generative-ai

* x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/results/attack_discovery_panel/tabs/alerts_tab/ai_for_soc/wrapper.tsx

`elastic/security-generative-ai` changes LGTM

christineweng committed:
Merge branch 'main' into alert-index

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311828)
* Commit: 6edc7d707c3bb07a2bb03728c6aa8bca886962f7

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/1d6a43944866562b7d9c429ccb8b10466edcebd5) | [after](https://github.com/elastic/kibana/commit/6edc7d707c3bb07a2bb03728c6aa8bca886962f7) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.4MB | 9.4MB | +1.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/1d6a43944866562b7d9c429ccb8b10466edcebd5) | [after](https://github.com/elastic/kibana/commit/6edc7d707c3bb07a2bb03728c6aa8bca886962f7) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 95.4KB | 95.4KB | +40.0B |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/1d6a43944866562b7d9c429ccb8b10466edcebd5) | [after](https://github.com/elastic/kibana/commit/6edc7d707c3bb07a2bb03728c6aa8bca886962f7) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 625 | 626 | +1 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/1d6a43944866562b7d9c429ccb8b10466edcebd5) | [after](https://github.com/elastic/kibana/commit/6edc7d707c3bb07a2bb03728c6aa8bca886962f7) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 343 | 345 | +2 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/1d6a43944866562b7d9c429ccb8b10466edcebd5) | [after](https://github.com/elastic/kibana/commit/6edc7d707c3bb07a2bb03728c6aa8bca886962f7) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 723 | 724 | +1 |


</details>


### History
* :green_heart: [Build #311723](https://buildkite.com/elastic/kibana-pull-request/builds/311723) succeeded 0f5482ffa9c0600b8cb6454437389f82d700e4ad
* :green_heart: [Build #311082](https://buildkite.com/elastic/kibana-pull-request/builds/311082) succeeded 0d4303935c373da0f0d529064c826ccfe066c4ff

cc @christineweng

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311828","number":311828,"commit":"6edc7d707c3bb07a2bb03728c6aa8bca886962f7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311723","number":311723,"commit":"0f5482ffa9c0600b8cb6454437389f82d700e4ad"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311082","number":311082,"commit":"0d4303935c373da0f0d529064c826ccfe066c4ff"}],"number":311828}
buildkite-pr-comment-->

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/225091">](https://github.com/elastic/kibana/pull/225091)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
